canopen/objectdictionary/__init__.py

```diff
@@ -185,7 +185,7 @@ def __repr__(self) -> str:
     def __getitem__(self, subindex: Union[int, str]) -> "ODVariable":
         item = self.names.get(subindex) or self.subindices.get(subindex)
         if item is None:
-            raise KeyError("Subindex %s was not found" % subindex)
+            raise KeyError(f"Subindex {subindex} was not found")
```

Comment:
I'd keep it as-is.  For integers, if there's no `0x` prefix, it can be assumed to be decimal.  Since this is only in an exception message, a developer will need to investigate anyway.